app/controllers/static_controller.rb

```diff
@@ -34,7 +34,7 @@ def set_featured_trainer
     return nil unless TeSS::Config.site.dig('home_page', 'featured_trainer')
 
     srand(Date.today.beginning_of_day.to_i)
-    Trainer.order(:id).sample(1)
+    Trainer.joins(:user).order(:id).filter { |t| t.user.image.present? }.sample(1)
```

Comment:
Try and avoid using filter unless its a small set of resources - it will load every trainer from the database into memory to do the filtering.

We check the `image_file_size` column in the provider carousel to only get providers with a logo: https://github.com/ElixirTeSS/TeSS/blob/f14b96d6602dbf09f6edf7b0c63c5b84740e86d9/app/views/static/home/_provider_carousel.html.erb#L5